src/custom_components/test_component.rs

```diff
@@ -0,0 +1,44 @@
+use ego_tree::NodeId;
+
+use crate::custom_component_renderer::dom_manipulator::{
+    create_node, create_text_node, NodeAttribute, NodeManipulator,
+};
+use crate::custom_component_renderer::error::RendererError;
+use crate::custom_component_renderer::error::Result;
+use crate::custom_component_renderer::languages_configuration::LanguagesConfiguration;
+use crate::custom_component_renderer::Component;
+
+pub struct TestComponent {}
+
+impl TestComponent {
+    pub fn new() -> TestComponent {
+        TestComponent {}
+    }
+}
+
+impl Component for TestComponent {
```

Comment:
Does this show that people need to write Rust code to introduce new components?

If so, that is problematic. The goal was for us/people to push logic into the templates and to avoid being constrained by the hard-coded functionality offered to the templates.